src/test/java/com/es/phoneshop/web/ProductListPageServletTest.java

```diff
@@ -37,5 +41,6 @@ public void testDoGet() throws ServletException, IOException {
         servlet.doGet(request, response);
 
         verify(requestDispatcher).forward(request, response);
+        verify(request).setAttribute(eq("products"), any());
```

Comment:
Try to mock ProductListDao and inject it into servlet with mockito, to test that list that set to "product" attribute is what provided from `ProductListDao.findProducts`